### PR TITLE
Author name links to author show page #28

### LINKED ISSUE #28: User Story 4, All author names are links to a show page
```
As a visitor
With the exception of an author's show page,
Anywhere I see an author's name on the site,
I can click on the name to go to that author's show page.
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written
- [ ] edge case code written
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #31

child of #33

depends on #14

### PR DESCRIPTION
**Pull request by milevy1**
Added author name links to their author show page.  Passes all specs with 100% coverage.

Closes #28

### CODE REVIEW

File: spec/features/book_show_spec.rb [hunk 1]
```diff
@@ -16,8 +16,8 @@
       expect(page).to have_content("Title: Where The Wild Things Are")
       expect(page).to have_content("Page Count: 40")
       expect(page).to have_content("Year Published: 1987")
-      expect(page).to have_content("Author 1 name")
-      expect(page).to have_content("Author 2 name")
+      expect(page).to have_link("Author 1 name", href: author_path(author_1))
+      expect(page).to have_link("Author 2 name", href: author_path(author_2))
```
Comment (Vjp888) [inline spec/features/book_show_spec.rb:20]: I did not know about being able to call the href in the expects. That's going to be really handy.

**Comment by Vjp888** [comment]
Everything looks good, I placed a comment but I like the href call in the expects, didn't know that was available. 

I approve for merge.
